go/cmd/dolt/commands/schcmds/tags.go

```diff
@@ -87,24 +88,63 @@ func (cmd TagsCmd) Exec(ctx context.Context, commandStr string, args []string, d
 			return 0
 		}
 	}
-	for i := 0; i < len(tables); i++ {
-		tables[i] = fmt.Sprintf("'%s'", tables[i])
+
+	var headerSchema = sql.Schema{
+		{Name: "table", Type: sql.Text, Default: nil},
+		{Name: "column", Type: sql.Text, Default: nil},
+		{Name: "tag", Type: sql.Uint64, Default: nil},
+	}
+
+	rows := make([]sql.Row, 0)
+
+	for _, tableName := range tables {
+		table, foundTableKey, ok, err := root.GetTableInsensitive(ctx, tableName)
+
+		// Return an error if table is not found
+		if !ok {
+			return commands.HandleVErrAndExitCode(errhand.BuildDError("Can't find table %s.", tableName).AddCause(err).Build(), usage)
+		}
+
+		if err != nil {
+			return commands.HandleVErrAndExitCode(errhand.BuildDError("Could not load table %s.", tableName).AddCause(err).Build(), usage)
+		}
+
+		sch, err := table.GetSchema(ctx)
+
+		if err != nil {
+			return commands.HandleVErrAndExitCode(errhand.BuildDError("Could not load %s schema.", tableName).AddCause(err).Build(), usage)
+		}
+
+		_ = sch.GetAllCols().Iter(func(tag uint64, col schema.Column) (stop bool, err error) {
+			rows = append(rows, sql.NewRow(foundTableKey, col.Name, tag))
+
+			return false, err
+		})
+
+	}
+
+	outputFmt, verr := commands.GetResultFormat("tabular")
+
+	if verr != nil {
+		return commands.HandleVErrAndExitCode(errhand.VerboseErrorFromError(verr), usage)
 	}
 
-	//TODO: implement REGEXP_SUBSTR in go-mysql-server and use it here instead of SUBSTR, as this will eventually break
-	queryStr := fmt.Sprintf("SELECT table_name AS 'table', column_name AS 'column', "+
-		"SUBSTR(extra, 5) AS tag FROM information_schema.columns WHERE table_name IN (%s)", strings.Join(tables, ","))
-
-	if formatStr, ok := apr.GetValue(commands.FormatFlag); ok {
-		return commands.SqlCmd{}.Exec(ctx, "", []string{
-			fmt.Sprintf(`--%s=%s`, commands.FormatFlag, formatStr),
-			fmt.Sprintf(`--%s`, commands.QueryFlag),
-			queryStr + ";",
-		}, dEnv)
-	} else {
-		return commands.SqlCmd{}.Exec(ctx, "", []string{
-			fmt.Sprintf(`--%s`, commands.QueryFlag),
-			queryStr + ";",
-		}, dEnv)
+	formatSr, ok := apr.GetValue(commands.FormatFlag)
+
+	var err error
+	if ok {
+		outputFmt, verr = commands.GetResultFormat(formatSr)
+
+		if verr != nil {
+			return commands.HandleVErrAndExitCode(errhand.VerboseErrorFromError(verr), usage)
+		}
 	}
+
+	err = commands.PrettyPrintResults(ctx, outputFmt, headerSchema, sql.RowsToRowIter(rows...))
+
+	if err != nil {
```

Comment:
Since this is the end of the function, you can just skip the condition and always return HandleVerrAndExitCode (it handles a nil verr correctly)